src/Umbraco.Infrastructure/HostedServices/TempFileCleanup.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Umbraco.Core;
+using Umbraco.Core.IO;
+
+namespace Umbraco.Infrastructure.HostedServices
+{
+    /// <summary>
+    /// Used to cleanup temporary file locations.
+    /// </summary>
+    /// <remarks>
+    /// Will run on all servers - even though file upload should only be handled on the master, this will
+    /// ensure that in the case it happes on replicas that they are cleaned up too.
```

Comment:
I know this comment is not new, but it seems like it is wrong.. I only runs if server is main dom

Reply:
I think it may still be valid.  If you look at other scheduled tasks - e.g. `KeepAlive` - these have checks via `_serverRegistrar.GetCurrentServerRole()` to ensure they don't run on replica servers.  This particular task doesn't have that check though, so seems this comment is there to call that out.